src/hist/basehist.py

```diff
@@ -94,6 +94,12 @@ def __init__(
                 warnings.warn(msg)
                 storage = storage()
             super().__init__(*args, storage=storage, metadata=metadata)  # type: ignore
+
+            disallowed_names = ["weight", "sample", "threads"]
+            if any(item in disallowed_names for item in [ax.name for ax in self.axes]):
+                disallowed_warning = "You have used a disallowed name for axes"
```

Comment:
```suggestion
                disallowed_warning = f"{item} is a protected keyword and cannot be used as axis name"
```

Are we past 3.6 yet?

Reply:
We'll drop 3.6 after this release.